libkbfs/path.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
```

Comment:
We have a `path` type in `data_types.go` already.  Maybe it makes sense to move that in here (along with `pathNode`), or else rename this file to `canonical_path.go`?